pkg/kubelet/kubelet.go

```diff
@@ -368,21 +364,19 @@ func NewMainKubelet(
 	}
 	klet.containerManager = containerManager
 
-	go util.Until(klet.syncNetworkStatus, 30*time.Second, util.NeverStop)
-	if klet.kubeClient != nil {
-		// Start syncing node status immediately, this may set up things the runtime needs to run.
```

Comment:
As we discussed offline, this PR should not change the existing startup
sequence. I am running e2e a few times to validate my assertion.

On Fri, Sep 25, 2015 at 11:47 AM, Yu-Ju Hong <notifications@github.com>
wrote:

> In pkg/kubelet/kubelet.go
> <https://github.com/kubernetes/kubernetes/pull/14542#discussion_r40462969>
> :
>
> > @@ -368,21 +364,19 @@ func NewMainKubelet(
> >  	}
> >  	klet.containerManager = containerManager
> >
> > -	go util.Until(klet.syncNetworkStatus, 30*time.Second, util.NeverStop)
> > -	if klet.kubeClient != nil {
> > -		// Start syncing node status immediately, this may set up things the runtime needs to run.
>
> Hmm...I think it's like this:
>
>    - syncNodeStatus gets the podCIDR from the Node object, and sets it.
>    - syncNetworkStatus configures the cbr0 bridge based on podCIDR, and
>    restart docker if necessary
>    - wait for runtime to be up
>
> It's not very clear in the current code that this will always happen in
> the desired order through, so I think it may not happen in the first try,
> but will eventually succeed.
>
> Given that you are changing the order, it's the impossible that this may
> caused unexpected situation to happen: longer startup time or restarting
> docker much later (at syncPod). It may worth spend some extra time making
> sure this would not happen with your PR.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/14542/files#r40462969>.
>